### PR TITLE
PR: Add support for Device Info properties: Software Revision, System ID, PnP ID

### PR DESCRIPTION
**Pull request by cpatterson**
The project I'm working on needs to support the Device Info Software Revision and System ID properties. This code modifies `RZBDeviceInfo` to support those properties, as well as PnP ID.

Note that System ID and PnP ID are binary values, not strings, so I created `RZBSystemId` and `RZBPnPId` classes to encapsulate that data.

`RZBDeviceInfo` unit tests were updated as well.

### CODE REVIEW

File: RZBluetooth/Profiles/RZBDeviceInfo.h [hunk 1]
```diff
@@ -9,6 +9,42 @@
...
+
+/// 40-bit Manufacturer-defined identifier
+@property (assign, nonatomic) UInt64 manufacturerId;
+/// 24-bit Organizationally Unique Identifier aka Company Identifier assigned by IEEE
+@property (assign, nonatomic) UInt32 ouid;
+
+@end
+
+/// Options for RZBPnPId.vendorIdSource property
+typedef enum _RZBVendorIdSource : UInt8 {
```
Comment (KingOfBrian) [inline RZBluetooth/Profiles/RZBDeviceInfo.h]: Any reason not to use the `NS_ENUM` system? I'd prefer to use it if possible. Also, I'd prefer to have the values follow the prefixed pattern -- ie: `RZBVendorIdSourceReserved`.

File: RZBluetooth/Profiles/RZBDeviceInfo.m [hunk 2]
```diff
@@ -35,17 +144,51 @@ + (CBCharacteristicProperties)characteristicPropertiesForKey:(NSString *)key
 
 + (id)valueForKey:(NSString *)key fromData:(NSData *)data
 {
-    return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+    if ([key isEqualToString:@"systemId"]) {
+        return [RZBSystemId valueForKey:key fromData:data];
+    }
+    else
+    if ([key isEqualToString:@"pnpId"]) {
```
Comment (KingOfBrian) [inline RZBluetooth/Profiles/RZBDeviceInfo.m]: Can you consolidate the `else` and `if`?  ie: `else if () {` -- Here and a few other places below.

Comment (cpatterson) [inline RZBluetooth/Profiles/RZBDeviceInfo.m]: 👍 No problem. Personal preference. 

I'll update the PR with these changes as soon as I can get back to it, probably tomorrow.

File: RZBluetooth/Profiles/RZBDeviceInfo.h (hunk 1 above)
Comment (cpatterson) [inline RZBluetooth/Profiles/RZBDeviceInfo.h]: 👍 No reason other than it's been too long since I wrote Obj-C code! :)

**Comment by cpatterson-lilly** [comment]
@KingOfBrian I've pushed another commit to my branch with the changes you requested.

**Comment by KingOfBrian** [comment]
Thanks @cpatterson -- I should have a new release out later today with this issue resolved and another more important callback bug. Thanks for taking the time to push up some improvements!

I'm going to give travis a few more hours to get to testing things, if it doesn't run I'll verify locally before the merge. Think I should move onto Circle, the free tier should get to a build in under a few hours, and this has been sitting for days...

**Pull request merged**